### ISSUE TITLE
About datasets

### ISSUE THREAD

**Issue by beijixiong3510** [daoyuan98/MDDA#2]
“we preprocess all of these data into a json file ”
Thank you for your open source code. May I ask if the processed data set in jison format can be uploaded, or tell me how to handle it.

**Comment by dyyin** [comment]
you can use image dataset to train this code. use below two function to replace dataset.py

`def get_files(filename):
    class_train = []
    label_train = []
    for train_class in os.listdir(filename):
        for pic in os.listdir(filename+train_class):
            class_train.append(filename+train_class+'/'+pic)
            label_train.append(train_class)
    temp = np.array([class_train,label_train])
    print(temp)
    temp = temp.transpose()
    np.random.shuffle(temp)
    image_list = list(temp[:,0])
    label_list = list(temp[:,1])
    label_list = [int(i) for i in label_list]
    return image_list,label_list


def get_batch(image, label, image_W, image_H, batch_size, capacity):
    image = tf.cast(image, tf.string)  
    label = tf.cast(label, tf.int32)
    input_queue = tf.train.slice_input_producer([image, label])
    label = input_queue[1]
    image_contents = tf.read_file(input_queue[0])  
    image = tf.image.decode_jpeg(image_contents, channels=3)

    image = tf.image.resize_image_with_crop_or_pad(image, image_W, image_H)
    image = tf.image.per_image_standardization(image)
    image_batch, label_batch = tf.train.batch([image, label], batch_size=batch_size, num_threads=16, capacity=capacity)
    label_batch = tf.reshape(label_batch, [batch_size])
    # image_batch = tf.cast(image_batch, tf.uint8)    
    image_batch = tf.cast(image_batch, tf.float32)  
    # print(label_batch) Tensor("Reshape:0", shape=(6,), dtype=int32)
    return image_batch, label_batch
`

**Comment by LisaWang0306** [comment]
Did you solve this problem? I am confused about it, too.

**Comment by beijixiong3510** [comment]
没有。。。



---Original---
From: ***@***.***&gt;
Date: Sat, Oct 23, 2021 13:23 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [daoyuan98/MDDA] About datasets (#2)




 
Did you solve this problem? I am confused about it, too.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by AIJIA000** [comment]
> Did you solve this problem?



> 

Did you solve this problem?

**Comment by jiang18-318** [comment]
i have the same question
